### ISSUE TITLE
Use same data types to build and to parse commands

### ISSUE THREAD

**Issue by crespum** [diondokter/at-commands#8]
I've defined some AT commands in local constants:
```rust
const CMD_PIN_QUERY: &[u8] = b"+CPIN";
const CMD_PIN_READY: &[u8] = b"+CPIN: READY";
```

And I'd like to use them like this:
```rust
let result = CommandBuilder::create_query(&mut buffer, true)
    .named(CMD_PIN_QUERY)
    .finish()
    .unwrap();

// TODO Send request
// TODO Receive response

match CommandParser::parse(&buffer)
    .expect_identifier(CMD_PIN_READY)
    .unwrap();
```
However this gives me the following error:
```
error[E0308]: mismatched types
   --> examples/../src/modem.rs:177:20
    |
177 |             .named(CMD_PIN_QUERY)
    |              ----- ^^^^^^^^^^^^^ expected `&str`, found `&[u8]`
    |              |
    |              arguments to this method are incorrect
    |
    = note: expected reference `&str`
               found reference `&'static [u8]`
```

Wouldn't it be _easier_ if both functions used the same parameter type (`&[u8]`)?

**Comment by diondokter** [comment]
If we can keep it backwards compatible, then sure. Should be doable by using something like `AsRef<[u8]>` which is implemented by `&str`

**Comment by diondokter** [comment]
Closed by #10
